[shopsys] move data fixtures to project-base (#854)
shopsys/shopsys: docs/upgrade/UPGRADE-unreleased.md

- *(low priority)* add `product-search-export-products` as a dependency of `build-demo` phing target in your `build.xml`
if you want to have products data exported to Elasticsearch after `build-demo` target is run ([#824](https://github.com/shopsys/shopsys/pull/824/files))

### Application

> - moved data fixtures to project-base [#854](https://github.com/shopsys/shopsys/pull/854)

Reviewer: from the example above, in UPGRADES there are now PRs in additional braces and they contain only instructions so
```suggestion
- copy data fixtures into your project namespace `ShopBundle\DataFixtures` from `FrameworkBundle\DataFixtures` of [shopsys/framework] repository ([#854](https://github.com/shopsys/shopsys/pull/854))
```
or something like that.

    - moving data fixtures was done in this [#854](https://github.com/shopsys/shopsys/pull/854) so you can get inspiration how to do it

>         - you need to move all data fixtures to a new folder `DataFixtures` in `ShopBundle`

Reviewer: OK so from where to where

Author: I removed this line because it's said in the previous point

if you want to have products data exported to Elasticsearch after `build-demo` target is run ([#824](https://github.com/shopsys/shopsys/pull/824/files))

### Application
- copy data fixtures into your project namespace `ShopBundle\DataFixtures` from `FrameworkBundle\DataFixtures` of [shopsys/framework] repository ([#854](https://github.com/shopsys/shopsys/pull/854))

>     - moving data fixtures was done in [#854](https://github.com/shopsys/shopsys/pull/854) so you can get inspiration how to do it

Reviewer: ```suggestion
    - moving data fixtures was done in ([#854](https://github.com/shopsys/shopsys/pull/854)) so you can get inspiration how to do it
```

Reviewer: i think this is the same instruction as above

>         - move configuration file `Resources/config/services/data_fixtures.yml` and import it in `services.yml` like this:

Reviewer: move it where ?

Author: The information was added in later commits

        ```diff
            imports:
                - { resource: forms.yml }
        +       - { resource: services/data_fixtures.yml }
        ```
        - change namespaces of data fixtures from `Shopsys\FrameworkBundle\DataFixtures` to `Shopsys\ShopBundle\DataFixtures` (in data fixtures, tests and *.yml configurations)

>         - add skipping of 4 data fixture files to `easy-coding-standards.yml`

Reviewer: this could be tricky for @Miroslav-Stopka but i think that he will know that easy-coding-standards.yml file is in the root of the project :100:

Author: I think its understandable, but I'll discuss it on BV

### Application
- *(low priority)* if you want to test behavior of cart with no listable product in it, implement functional test as it is in [#852](https://github.com/shopsys/shopsys/pull/852)

> - copy data fixtures into your project namespace `ShopBundle\DataFixtures` from `FrameworkBundle\DataFixtures` of [shopsys/framework] repository ([#854](https://github.com/shopsys/shopsys/pull/854))

Reviewer 2: That is not true, unfortunately. 
Imagine a situation where I have a project on Beta6 (Data fixtures are part of Framework). I update dependencies and start upgrading and get to this instruction.
I look into my vendor/shopsys/framework for data and they're not there. 
I look at Github to shopsys framework repository and they're not there either.

They had to copy them from project-base. Same applies for all following instructions.

    - copy `PerformanceDataCommand.php` into project namespace `ShopBundle\Command` from `FrameworkBundle\Command` of [shopsys/framework] repository
    - move configuration file `FrameworkBundle/Resources/config/services/data_fixtures.yml` to `Shopbundle/Resources/config/services/data_fixtures.yml`
    - add new configuration file to `Shopbundle/Resources/config/services/commands.yml` and add project-base Command configuration
    ```yaml
    services:
        _defaults:
            autowire: true
            autoconfigure: true
            public: false

        Shopsys\ShopBundle\Command\:
            resource: '../../../Command'
    ```
    - import both configuration files in `services.yml`
    ```diff
        imports:
            - { resource: forms.yml }
    +       - { resource: 'services/{commands,data_fixtures}.yml' }
    ```

>     - change namespaces of data fixtures from `Shopsys\FrameworkBundle\DataFixtures` to `Shopsys\ShopBundle\DataFixtures` (in data fixtures, tests and *.yml configurations)

Reviewer 2: If I copy datafixtures from shopbundle I don't need to change namespaces in datafixtures. 
Also I'm missing some instructions on what to do when I have some datafixtures extended. Take a look at demoshop and try to imagine you have to update datafixtures there (mail templates) based on this informations only.

### Application
- *(low priority)* if you want to test behavior of cart with no listable product in it, implement functional test as it is in [#852](https://github.com/shopsys/shopsys/pull/852)
- copy data fixtures into your project namespace `ShopBundle/DataFixtures` from [`src/Shopsys/ShopBundle/DataFixtures`](https://github.com/shopsys/project-base/tree/master/src/Shopsys/ShopBundle/DataFixtures) of [shopsys/project-base] repository ([#854](https://github.com/shopsys/shopsys/pull/854))
    - copy `PerformanceDataCommand.php` into project namespace `ShopBundle/Command` from [`src/Shopsys/ShopBundle/Command`](https://github.com/shopsys/project-base/tree/master/src/Shopsys/ShopBundle/Command) of [shopsys/project-base] repository

>     - copy configuration file to `Shopbundle/Resources/config/services/data_fixtures.yml` from [`src/Shopsys/ShopBundle/Resources/config/services/data_fixtures.yml`](https://github.com/shopsys/project-base/tree/master/src/Shopsys/ShopBundle/Resources/config/services/data_fixtures.yml) of [shopsys/project-base] repository

Reviewer 2: ```suggestion
    - copy configuration file to `ShopBundle/Resources/config/services/data_fixtures.yml` from [`src/Shopsys/ShopBundle/Resources/config/services/data_fixtures.yml`](https://github.com/shopsys/project-base/tree/master/src/Shopsys/ShopBundle/Resources/config/services/data_fixtures.yml) of [shopsys/project-base] repository
```

>     - add new configuration file to `Shopbundle/Resources/config/services/commands.yml` and add Command configuration in it

Reviewer 2: ```suggestion
    - add new configuration file to `ShopBundle/Resources/config/services/commands.yml` and add Command configuration in it
```
